patches/api/0384-Event-Effect-API-Improvements.patch

```diff
@@ -0,0 +1,775 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Sun, 5 Jun 2022 21:19:40 -0400
+Subject: [PATCH] Event/Effect API Improvements
+
+
+diff --git a/src/main/java/io/papermc/paper/block/event/BlockEvent.java b/src/main/java/io/papermc/paper/block/event/BlockEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..d4e693581127862c35d1ece432458d2eeeaa6abd
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/block/event/BlockEvent.java
+@@ -0,0 +1,31 @@
++package io.papermc.paper.block.event;
++
++import org.bukkit.Material;
++import org.jetbrains.annotations.ApiStatus;
++
++import java.util.Set;
++
++/**
++ * Represents an event that will cause
++ * an effect to occur for a block.
++ */
++public abstract class BlockEvent {
```

Comment:
`@ApiStatus.NonExtendable`

or just make it a concrete type that is sealed to inner `Configurable` type (or a better name).